Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -816,23 +818,28 @@ void BlockFormattingContext::layout_block_level_box(Box const& box, BlockContain
         resolve_used_height_if_treated_as_auto(box, available_space_for_height_resolution);
     }
 
+    // This monster basically means: "a ListItemBox that does not have specified content in the ::marker pseudo-element".
+    // This happens for ::marker with content 'normal'.
+    bool is_list_item_box_without_css_content = is<ListItemBox>(box) && (!(box.dom_node() && box.dom_node()->is_element() && static_cast<DOM::Element const*>(box.dom_node())->computed_properties(CSS::PseudoElement::Marker)->property(CSS::PropertyID::Content).is_content()));
+
```

Comment:
it seems you’re using `is<ListItemBox>(box)` below in form of `auto const* li_box = as_if<ListItemBox>(box)`. maybe it’d be better to start with that here and use below, and define `is_list_item_box_without_css_content` in terms of `li_box`? (i guess that you can also assume, that `dom_node()` return value exists, all these would probably significantly simplify this “monster” as you call it.)